clients/src/main/java/org/apache/kafka/clients/consumer/internals/metrics/AsyncConsumerMetrics.java

```diff
@@ -0,0 +1,262 @@
+    private final Sensor applicationEventExpiredSizeSensor;
+    private final Sensor backgroundEventQueueSizeSensor;
+    private final Sensor backgroundEventQueueTimeSensor;
+    private final Sensor backgroundEventQueueProcessingTimeSensor;
+    private final Sensor unsentRequestsQueueSizeSensor;
+    private final Sensor unsentRequestsQueueTimeSensor;
+
+    public AsyncConsumerMetrics(Metrics metrics) {
+        super(metrics, CONSUMER_METRIC_GROUP_PREFIX);
+
+        this.metrics = metrics;
+        this.timeBetweenNetworkThreadPollSensor = metrics.sensor(TIME_BETWEEN_NETWORK_THREAD_POLL_SENSOR_NAME);
+        this.timeBetweenNetworkThreadPollSensor.add(
+            metrics.metricName(
+                "time-between-network-thread-poll-avg",
+                CONSUMER_METRIC_GROUP,
+                "The average time taken, in milliseconds, between each poll in the network thread."
+            ),
+            new Avg()
+        );
+        this.timeBetweenNetworkThreadPollSensor.add(
+            metrics.metricName(
+                "time-between-network-thread-poll-max",
+                CONSUMER_METRIC_GROUP,
+                "The maximum time taken, in milliseconds, between each poll in the network thread."
+            ),
+            new Max()
+        );
+
+        this.applicationEventQueueSizeSensor = metrics.sensor(APPLICATION_EVENT_QUEUE_SIZE_SENSOR_NAME);
+        this.applicationEventQueueSizeSensor.add(
+            metrics.metricName(
+                APPLICATION_EVENT_QUEUE_SIZE_SENSOR_NAME,
+                CONSUMER_METRIC_GROUP,
+                "The current number of events in the queue to send from the application thread to the background thread."
+            ),
+            new Value()
+        );
+
+        this.applicationEventQueueTimeSensor = metrics.sensor(APPLICATION_EVENT_QUEUE_TIME_SENSOR_NAME);
+        this.applicationEventQueueTimeSensor.add(
+            metrics.metricName(
+                "application-event-queue-time-avg",
+                CONSUMER_METRIC_GROUP,
+                "The average time, in milliseconds, that application events are taking to be dequeued."
+            ),
+            new Avg()
+        );
+        this.applicationEventQueueTimeSensor.add(
+            metrics.metricName(
+                "application-event-queue-time-max",
+                CONSUMER_METRIC_GROUP,
+                "The maximum time, in milliseconds, that an application event took to be dequeued."
+            ),
+            new Max()
+        );
+
+        this.applicationEventQueueProcessingTimeSensor = metrics.sensor(APPLICATION_EVENT_QUEUE_PROCESSING_TIME_SENSOR_NAME);
+        this.applicationEventQueueProcessingTimeSensor.add(
+            metrics.metricName(
+                "application-event-queue-processing-time-avg",
+                CONSUMER_METRIC_GROUP,
+                "The average time, in milliseconds, that the consumer network takes to process all available application events."
```

Comment:
```suggestion
                "The average time, in milliseconds, that the background thread takes to process all available application events."
```
background thread or "network thread" is fine (but not "consumer network"). Let's just be consistent and use the same term when possible in all metrics